hugo-symbol-corrector/gene_symbol_correction.py

```diff
@@ -1,42 +1,72 @@
-import sys
+import argparse
 import os
 
-gene_dict = {}
-with open(sys.argv[1],"r") as mp:
-	for line in mp:
-		line = line.rstrip('\n')
-		genes = line.split("\t")
-		gene_dict[genes[1]] = genes[0]
+parser = argparse.ArgumentParser()
+parser.add_argument("--mapping", dest='mapping', help="This is the mapping file path")
+parser.add_argument("--hugo", dest='hugo', help="This is the file path which contains all the hugo symbols that needs to be replaced")
+args = parser.parse_args()
+
+
+if __name__ == "__main__":
+	gene_dict = {} # this has the keys of all the gene mappings
+	header_dict = {} # this has all the keys of column names and their respective array ids
+ 
+	# reading the mapping file and saving to dict
+	with open(args.mapping ,"r") as mp:
+		for line in mp:
+			line = line.rstrip('\n')
+			genes = line.split("\t")
+			gene_dict[genes[1]] = genes[0]
 	
-comments = ""
-header = ""
-data = ""
-replaced_list = {}
-
-with open(sys.argv[2],"r") as file:
-	for line in file:
-		if line.startswith('#'):
-			comments += line
-		elif line.startswith('Hugo_Symbol'):
-			header += line
-		else:
-			values = line.split("\t")
-			values[0] = values[0].strip('\t')
-			if values[0].upper() in gene_dict:
-				line = line.replace(values[0],gene_dict[values[0].upper()],1)
-				replaced_list[values[0]] = gene_dict[values[0].upper()]
-				data += line
+
+	comments = ""
+	data = ""
+	columns = ""
+	replaced_list = {}
+
+	# reading the hugo symbols file
+	with open(args.hugo ,"r") as file:
+		counter = 0
+		for i, line in enumerate(file):
+			# this condition is to get all the column names and save it in header_dict  
+			if i == 0:
+				columns += line
+				header = line.split("\t") 
+				for i, head in enumerate(header):
+					head = head.rstrip('\n')
+					header_dict[head] = i
+				continue
+			if line.startswith('#'):
+				comments += line
 			else:
-				data += line	
+				replace_line = None
+				# looping through the required column names
+				for gene_name in ["Site1_Hugo_Symbol", "Site2_Hugo_Symbol", "Hugo_Symbol"]:
```

Comment:
Good to also check Event_Info field for SV?

Reply:
Can we also include RPPA and protein expression files? The genes are stored in `Composite.Element.REF`. We will have to split by `|` for antibodies and check for genes.